package.json

```diff
@@ -1,808 +1,58 @@
-    "test-compile": "tsc -p ./",
-    "compile": "tsc -p ./",
-    "watch": "tsc -watch -p ./",
-    "pretest": "yarn run compile && yarn run lint && yarn run esbuild",
-    "lint": "eslint packages --ext ts",
-    "test": "env CURSORLESS_TEST=true node ./out/test-harness/scripts/runTestsCI.js",
+    "preinstall": "npx only-allow pnpm",
+    "test-compile": "tsc --build",
+    "build": "pnpm compile && pnpm -r build",
+    "compile": "tsc --build",
+    "watch": "tsc --build -watch",
+    "meta-updater:base": "pnpm --filter=@cursorless/meta-updater compile && meta-updater",
+    "meta-updater": "pnpm run meta-updater:base && pnpm -r exec prettier --write tsconfig.json package.json",
+    "lint:meta": "pnpm run meta-updater:base --test",
+    "lint": "pnpm run lint:meta && syncpack list-mismatches && pnpm run lint:ts",
+    "lint:ts": "eslint packages --ext ts",
+    "test": "pnpm compile && pnpm lint && pnpm -F '!test-harness' test && pnpm -F test-harness test",
+    "transform-recorded-tests": "pnpm -F @cursorless/cursorless-engine transform-recorded-tests",
     "unused-exports": "ts-unused-exports tsconfig.json --showLineNumber",
-    "init-launch-sandbox": "node ./out/cursorless-vscode-core/scripts/initLaunchSandbox.js",
-    "prepare-for-extension-publish": "node ./out/cursorless-vscode-core/scripts/prepareForExtensionPublish.js",
-    "transform-recorded-tests": "node ./out/cursorless-engine/scripts/transformRecordedTests/index.js"
+    "update-licenses": "npx npm-license-crawler --onlyDirectDependencies --csv third-party-licenses.csv"
   },
+  "keywords": [],
+  "author": "",
+  "license": "MIT",
   "devDependencies": {
-    "@types/chai": "^4.3.3",
-    "@types/glob": "^7.1.3",
-    "@types/js-yaml": "^4.0.2",
-    "@types/lodash": "^4.14.168",
-    "@types/mocha": "^8.0.4",
+    "@pnpm/meta-updater": "0.2.2",
     "@types/node": "^16.11.3",
-    "@types/semver": "^7.3.9",
-    "@types/sinon": "^10.0.2",
-    "@types/uuid": "^8.3.4",
-    "@types/vscode": "~1.61.0",
     "@typescript-eslint/eslint-plugin": "^5.20.0",
     "@typescript-eslint/parser": "^5.20.0",
-    "@vscode/test-electron": "^2.1.3",
-    "chai": "^4.3.6",
-    "esbuild": "^0.11.12",
+    "eslint": "^8.13.0",
     "eslint-config-prettier": "^8.5.0",
+    "eslint-import-resolver-typescript": "3.5.3",
+    "eslint-plugin-import": "2.27.5",
     "eslint-plugin-unused-imports": "^2.0.0",
-    "eslint": "^8.13.0",
-    "fast-xml-parser": "^3.20.0",
-    "glob": "^7.1.7",
-    "js-yaml": "^4.1.0",
-    "mocha": "^8.1.3",
-    "module-alias": "^2.2.2",
-    "npm-license-crawler": "^0.2.1",
     "prettier": "2.7.1",
-    "semver": "^7.3.7",
-    "sinon": "^11.1.1",
-    "ts-unused-exports": "^8.0.0",
-    "typescript": "4.6.3"
+    "syncpack": "9.8.4",
+    "tsx": "3.12.3",
+    "typescript": "^4.9.5"
+  },
+  "pnpm": {
+    "patchedDependencies": {
+      "@docusaurus/theme-search-algolia@2.3.1": "patches/@docusaurus__theme-search-algolia@2.3.1.patch"
```

Comment:
file an issue to figure out what happens when this gets upgraded.  will it just silently stop using patch?